src/ghostwriter/hooks/github_notebook.py

```diff
@@ -9,14 +9,11 @@
 from string import Template
 from urllib.parse import urljoin
 
-import structlog
-
 from ..models.substitution import Parameters
-
-LOGGER = structlog.get_logger("ghostwriter")
+from ._logger import LOGGER
 
 
-async def github_notebook(params: Parameters) -> Parameters | None:
+async def github_notebook(params: Parameters) -> Parameters:
```

Comment:
Perhaps a suggestion for when you have more development effort on this would be splitting the path processing here to its own method and writing some tests for that to make sure it can handle all unusual cases of paths